controller/engine_controller.go

```diff
@@ -1653,10 +1653,15 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 	}
 	// replica has already been added to the engine
 	if alreadyExists {
-		ec.logger.Debugf("Replica %v address %v has been added to the engine already", replica, addr)
+		ec.logger.Debugf("Replica %v address %v has been added to the engine already", replicaName, addr)
 		return nil
 	}
 
+	replica, err := ec.ds.GetReplica(replicaName)
+	if err != nil {
+		return err
+	}
```

Comment:
This replica is used in the below goroutine only, so let's move into it and just before `ec.updateReplicaRebuildFailedCondition(replica, "")`.